zipkin-ui/build.gradle

```diff
@@ -0,0 +1,45 @@
+apply plugin: 'com.moowork.node'
+
+artifacts.zipkinUpload jar
+
+sourceSets {
+    main {
+        resources {
+            srcDir 'static'
+            srcDir 'dist'
+        }
+    }
+}
+clean {
+    delete "dist"
+}
+node {
+    // Version of node to use.
+    version = '5.5.0'
+
+    // Base URL for fetching node distributions (change if you have a mirror).
+    distBaseUrl = 'https://nodejs.org/dist'
+
+    // If true, it will download node using above parameters.
+    // If false, it will try to use globally installed node.
+    download = true
+
+    // Set the work directory for unpacking node
+    workDir = file("${project.buildDir}/nodejs")
+
+    // Set the work directory where node_modules should be located
+    nodeModulesDir = file("${project.projectDir}")
+}
+
+task webpack(type: NpmTask) {
+    args = ['run-script', 'build']
+}
+task karma(type: NpmTask) {
+    args = ['run-script', 'test']
+}
+webpack.dependsOn(npmInstall)
+karma.dependsOn(npmInstall)
+build.dependsOn(webpack)
+build.dependsOn(karma)
+processResources.dependsOn(webpack)
+processResources.dependsOn(karma)
```

Comment:
Here's how I'd set up the task dependencies. `check` is the Gradle standard "run all the tests and whatevers" task; `build` depends on it. I'm assuming the `karma` tests need the `webpack` build to have completed.

```
npmInstall ---> webpack ---> karma ---> check
                               |
                               + -----> processResources
```

This includes (either explicitly or transitively) all the dependencies you've specified, except for one: `karma -> processResources`. I'm guessing there's no actual dependency there, can you verify that? If there is, we can just move `processResources` to depend on `karma` instead of `webpack` (which depends on `webpack, so we still get it transitively)